src/ConnectedMode.UnitTests/Migration/ConnectedModeMigrationTests.cs

```diff
@@ -46,21 +47,93 @@ public void MefCtor_CheckIsExported_NoProjectCleaners()
         }
 
         [TestMethod]
-        public void Migrate_FilesExist_AllFilesCleaned()
+        public async Task Migrate_NoFilesToClean_DirectoryIsDeleted()
         {
-            var fileProvider = CreateFileProvider("file1", "file2", "file3");
+            var fileProvider = CreateFileProvider();
+            var fileSystem = new Mock<IVsAwareFileSystem>();
             var fileCleaner = new Mock<IFileCleaner>();
 
-            var testSubject = CreateTestSubject(fileProvider.Object, fileCleaner.Object);
+            var testSubject = CreateTestSubject(fileProvider.Object, fileCleaner.Object, fileSystem.Object);
 
-            Func<Task> act = async () => await testSubject.MigrateAsync(null, CancellationToken.None);
+            await testSubject.MigrateAsync(null, CancellationToken.None);
 
-            act.Should().NotThrow();
+            fileProvider.Verify(x => x.GetFilesAsync(It.IsAny<CancellationToken>()), Times.Once);
+
+            fileCleaner.Invocations.Should().HaveCount(0);
+            fileSystem.VerifyNoFilesSaved();
+            fileSystem.VerifyDirectoryDeleted("folder"); // TODO - check for the correct path. Depends on #4362
```

Comment:
`folder` is currently hard-coded because that bit of the product code has not been implemented yet.